src/main/java/Duke.java

```diff
@@ -1,10 +1,264 @@
+
+    /**
+     * Prints message within horizontal lines.
+     *
+     * @param message The message to print.
+     */
+    private static void printMessage(String message) {
+        System.out.println(MESSAGE_WALL);
+        System.out.println(message);
+        System.out.println(MESSAGE_WALL);
+    }
+
+    /**
+     * Prints tasks in list format.
+     *
+     * @param header List header.
+     */
+    private static void printTasks(String header) {
+        StringBuilder messageBuilder = new StringBuilder(header);
+        for (int i = 0; i < TASKS.size(); i += 1) {
+            messageBuilder.append("\n");
+            messageBuilder.append(i + 1);
+            messageBuilder.append(": ");
+            messageBuilder.append(TASKS.get(i).getListEntryString());
+        }
+
+        printMessage(messageBuilder.toString());
+    }
+
+    /**
+     * Checks if string value is an integer.
+     *
+     * @param string The string to check.
+     * @return true if string can be converted to integer, else false.
+     */
+    private static boolean isInteger(String string) {
+        boolean isInt;
+        try {
+            int value = Integer.parseInt(string);
+            isInt = true;
+        } catch (NumberFormatException e) {
+            isInt = false;
+        }
+
+        return isInt;
+    }
+
+    /**
+     * Gets a non-empty user input.
+     *
+     * @return Non-empty user input.
+     */
+    private static String getUserInput() {
+        String input;
+        do {
+            System.out.print(INPUT_PROMPT);
+            input = SCANNER.nextLine();
+        } while (input.trim().isEmpty());
+
+        return input;
+    }
+
+    /**
+     * Splits user input string into command and argument.
+     *
+     * @param input The user input string.
+     * @return String array: [0] - Command, [1] - argument.
+     */
+    private static String[] getCommandAndArgument(String input) {
+        String[] result = input.trim().split("\\s+",2);
+        return (result.length == 2) ? result : new String[] { result[0] , "" };
+    }
+
+    /**
+     * Executes the list command. Print tasks in the list.
+     */
+    private static void executeListCommand() {
+        if (TASKS.isEmpty()) {
+            printMessage(MESSAGE_LIST_EMPTY);
+        } else {
+            printTasks(MESSAGE_LIST_HEADER);
+        }
+    }
+
+    /**
+     * Checks if task index is valid.
+     * Note: Task index starts from 1 (not 0).
+     *
+     * @param taskIndex The task index.
+     * @return true if task index is valid, else false.
+     */
+    private static boolean isValidTaskIndex(int taskIndex) {
+        return (taskIndex > 0 && taskIndex <= TASKS.size());
+    }
+
+    /**
+     * Executes the done command. Marks the given task as done.
+     */
+    private static void executeDoneCommand(String argument) {
+        if (!isInteger(argument)) {
+            printMessage(String.format(MESSAGE_FORMAT_DONE_USAGE,COMMAND_DONE));
```

Comment:
Fix the indenting of this line. Make sure that all code within any method or class is positioned and indented properly. Basic indentation should be 4 spaces.